File: GestaoLivrosApi/GestaoLivrosApi/Helpers/MessagingHelper.cs

using Ardalis.SmartEnum;
using Ardalis.SmartEnum.SystemTextJson;
using System.Text.Json.Serialization;

namespace GestaoLivrosApi.Helpers
{
	public class MessagingHelper<T>
	{
        public bool Success { get; set; }
        public string Message { get; set; }
        public T Obj { get; set; }

        [JsonConverter(typeof(SmartEnumValueConverter<ErrorType, string>))]
        public ErrorType? ErrorType { get; set; } = null;
    }
    public class MessagingHelper
    {
        public bool Success { get; set; }
        public string Message { get; set; }

        [JsonConverter(typeof(SmartEnumValueConverter<ErrorType, string>))]
        public ErrorType? ErrorType { get; set; } = null;
    }

    public class ErrorType : SmartEnum<ErrorType, string>
    {
        public static readonly ErrorType DataHasChanged = new ErrorType("DataHasChanged");
        protected ErrorType(string name) : base(name, name) { }
    }
}

File: frontend/src/helpers/MessagingHelper.ts
export class MessagingHelper<T> {
    success: boolean;
    message: string;
    obj: T;
    errorType: ErrorType | null = null;

    constructor(success: boolean, message: string, obj: T) {
        this.success = success;
        this.message = message;
        this.obj = obj;
    }
}

export enum ErrorType {
    DataHasChanged = "DataHasChanged"
}